clients/roscpp/src/libros/xmlrpc_manager.cpp

```diff
@@ -155,21 +155,24 @@ void XMLRPCManager::shutdown()
   server_.close();
 
   // kill the last few clients that were started in the shutdown process
-  for (V_CachedXmlRpcClient::iterator i = clients_.begin();
-       i != clients_.end(); ++i)
   {
-    for (int wait_count = 0; i->in_use_ && wait_count < 10; wait_count++)
+    boost::mutex::scoped_lock lock(clients_mutex_);
+
+    for (V_CachedXmlRpcClient::iterator i = clients_.begin();
+         i != clients_.end(); ++i)
     {
-      ROSCPP_LOG_DEBUG("waiting for xmlrpc connection to finish...");
-      ros::WallDuration(0.01).sleep();
+      if (!i->in_use_)
+      {
+       i->client_->close();
+       delete i->client_;
+      }
     }
 
-    i->client_->close();
-    delete i->client_;
+    // Erase the deleted clients. The clients in use will delete themselves
+    // on release
+    clients_.erase(std::remove(clients_.begin(), clients_.end(), false), clients_.end());
```

Comment:
It looks like this file follows the 80 column maximum style. You need to break this in two lines.